API/projectClass/layersClassRealSLD.m

```diff
@@ -0,0 +1,293 @@
+            %                  parameters.paramsTable{:, 1});
+            switch length(whatToAdd)
+
+                case 0
+                    % Add an empty layer
+                    layerNum = obj.layersAutoNameCounter;
+                    layerName = sprintf('Layer %d',layerNum);
+                    newRow = {layerName,'','','','','bulk out'};
+                    appendNewRow(obj,newRow);
+                    
+                case 1
+                    % Add an empty named layer
+                    newRow = {whatToAdd{1},'','','','','bulk out'};
+                    appendNewRow(obj,newRow);
+                
+                otherwise
+                    % Add a layer that is fully defined
+                    layerDetails = whatToAdd;
+
+                    if length(layerDetails) == 4
+                        % No hydration
+                        layerDetails = {layerDetails{1},layerDetails{2},layerDetails{3},layerDetails{4},NaN,'bulk in'};
+                    elseif length(layerDetails) ~= 6
+                        error('Can''t define a layer from partial details')
+                    end
+                    
+                    name = layerDetails{1};
+                    hydrateWhat = layerDetails{end};
+                    
+                    if ~strcmpi(hydrateWhat,obj.allowedHydration)
+                        error('Hydrate type must be ''bulk in'', ''bulk out'' or ''none''');
+                    end
+                    
+                    % Check that the parameter names given are real
+                    % parameters or numbers
+                    thisRow = {name};
+                    
+                    % Must be a parameter name or number . . .
+                    for i = 2:4                       
+                        thisRow{i} = obj.findParameter(layerDetails{i}, paramNames);
+                    end
+
+                    %  . . . (unless p=5 which can also be Nan)
+                    if isnan(layerDetails{5})
+                        thisRow{5} = NaN;
+                    else
+                        thisRow{5} = obj.findParameter(layerDetails{5}, paramNames);
+                    end
+                    
+                    thisRow = [thisRow hydrateWhat];
+                    appendNewRow(obj,thisRow);
+                    
+            end
+            
+        end
+        
+        function obj = setLayerValue(obj, inputValues, paramNames)
+            % Change the value of a given layer parameter in the table
+            % (excluding the layer name). The row and column of the
+            % parameter can both be specified by either name or index.
+            % The expected input is a cell array consisting of three
+            % values: row, column, newValue, and a string array of
+            % parameter names defined in the project's parameter class.
+            %
+            % layers.setLayerValue(1, 1, "origin", parameters.paramsTable{:, 1});
```

Comment:
This example doesn't match the actual usage `layers.setLayerValue({1, 1, "origin"}, parameters.paramsTable{:, 1});`. I prefer the syntax in the docstring, you can use a varargin then take varargin{end} as the table then varargin{1:end-1} as the inputs. Another option would be to flip the argument order, you could do` function obj = setLayerValue(obj, paramNames, varargin)` which should allow you do `layers.setLayerValue(parameters.paramsTable{:, 1}, 1, 1, "origin");`

Reply:
The input has been changed in both the tests and the project class to match that in the docstring.